### PR TITLE
Add rule S881 for TS/JS (`no-nested-incdec` in tslint)

### LINKED ISSUE #1487: Add rule S881 for TS/JS (`no-nested-incdec` in tslint)

### PR DESCRIPTION
**Pull request by quentin-jaquier-sonarsource**
Fixes #1487

### CODE REVIEW

File: its/ruling/src/test/expected/ts/angular/typescript-S881.json [hunk 1]
```diff
@@ -147,10 +144,6 @@
 'angular:packages/core/testing/src/test_bed.ts':[
 442,
 ],
-'angular:packages/examples/common/ngIf/ts/module.ts':[
```
Comment (quentin-jaquier-sonarsource) [inline its/ruling/src/test/expected/ts/angular/typescript-S881.json]: Differences in angular are due to the fact that these are now ignored.

Comment (vilchik-elena) [reply its/ruling/src/test/expected/ts/angular/typescript-S881.json]: which "these" issues? what special about them that they are ignored?

File: eslint-bridge/tests/rules/no-nested-incdec.test.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    {
+      code: `foo[++i] = 0;`,
+      errors: 1,
+    },
+    {
+      code: `foo[--i] = 0;`,
+      errors: 1,
+    },
+    {
+      code: `() => {if (i == 1) {
```
Comment (vilchik-elena) [inline eslint-bridge/tests/rules/no-nested-incdec.test.ts]: what is the point of this big test?

File: eslint-bridge/src/rules/no-nested-incdec.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        switch (firstAncestor.type) {
+          case "ExpressionStatement":
+            return true;
+          case "ForStatement":
+            return firstAncestor.update === node;
+          case "SequenceExpression": {
+            const secondAncestor = ancestors.pop();
+            return (
+              secondAncestor !== undefined &&
+              secondAncestor.type === "ForStatement" &&
```
Comment (vilchik-elena) [inline eslint-bridge/src/rules/no-nested-incdec.ts]: I didn't find a test when this condition is false

File: eslint-bridge/tests/rules/no-nested-incdec.test.ts [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+    {
+      code: `foo[-i] = 0;`,
+    },
+    {
+      code: `let j = 0;
+            for (i = 0; i < 10; i++, j++, k++) {
+            }`,
+    },
+  ],
+  invalid: [
```
Comment (vilchik-elena) [inline eslint-bridge/tests/rules/no-nested-incdec.test.ts]: tbh these tests are a mess. I believe we should not mix tests with and without location assertion, it makes it unreadable. And it feels like these tests are redundant

File: its/ruling/src/test/expected/ts/angular/typescript-S881.json (hunk 1 above)
Comment (quentin-jaquier-sonarsource) [reply its/ruling/src/test/expected/ts/angular/typescript-S881.json]: I wanted to mean that the **files** in which the issues were appearing are now ignored during the ruling.

File: eslint-bridge/src/rules/no-nested-incdec.ts (hunk 3 above)
Comment (quentin-jaquier-sonarsource) [reply eslint-bridge/src/rules/no-nested-incdec.ts]: Test added

File: eslint-bridge/tests/rules/no-nested-incdec.test.ts (hunk 4 above)
Comment (quentin-jaquier-sonarsource) [reply eslint-bridge/tests/rules/no-nested-incdec.test.ts]: I agree, I re-organized the file.

**Pull request merged**